tests/components/amberelectric/test_services.py

```diff
@@ -0,0 +1,206 @@
+import pytest
+
+from homeassistant.components.amberelectric.const import (
+    CONF_SITE_ID,
+    CONF_SITE_NAME,
+    DOMAIN,
+    GET_FORECASTS_SERVICE,
+)
+from homeassistant.components.amberelectric.services import (
+    ATTR_CHANNEL_TYPE,
+    ATTR_SITE_ID,
+)
+from homeassistant.const import CONF_API_TOKEN
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ServiceValidationError
+
+from . import setup_integration
+from .helpers import FORECASTS, GENERAL_ONLY_SITE_ID
+
+from tests.common import MockConfigEntry
+
+
+async def create_amber_config_entry(site_id):
+    """Create an Amber config entry."""
+    return MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            CONF_API_TOKEN: "TOKEN",
+            CONF_SITE_NAME: "home",
+            CONF_SITE_ID: site_id,
+        },
+        entry_id=site_id,
+    )
+
+
+@pytest.fixture
+async def amber_config_entry():
+    """Generate the default Picnic config entry."""
+    return await create_amber_config_entry(GENERAL_ONLY_SITE_ID)
+
+
+@pytest.fixture(name="current_price_api")
+def mock_api_current_price() -> Generator:
+    """Return an authentication error."""
+    instance = Mock()
+
+    site = Site(
+        id=GENERAL_ONLY_SITE_ID,
+        nmi="11111111111",
+        channels=[
+            Channel(identifier="E1", type=ChannelType.GENERAL, tariff="A100"),
+            Channel(identifier="E2", type=ChannelType.CONTROLLEDLOAD, tariff="A180"),
+            Channel(identifier="B1", type=ChannelType.FEEDIN, tariff="A100"),
+        ],
+        network="Jemena",
+        status=SiteStatus("active"),
+        activeFrom=date(2021, 1, 1),
+        closedOn=None,
+        interval_length=30,
+    )
+
+    instance.get_sites = Mock(return_value=[site])
+    instance.get_current_prices = Mock(return_value=FORECASTS)
+
+    with patch("amberelectric.AmberApi", return_value=instance):
+        yield instance
+
+
+async def test_get_general_forecasts(
+    hass: HomeAssistant, current_price_api, amber_config_entry: MockConfigEntry
+) -> None:
+    """Test fetching general forecasts."""
+    await setup_integration(hass, amber_config_entry)
+    result = await hass.services.async_call(
+        DOMAIN,
+        GET_FORECASTS_SERVICE,
+        {ATTR_SITE_ID: GENERAL_ONLY_SITE_ID, ATTR_CHANNEL_TYPE: "general"},
+        blocking=True,
+        return_response=True,
+    )
+    assert len(result["forecasts"]) == 3
+
+    first = result["forecasts"][0]
+    assert first["duration"] == 30
+    assert first["date"] == "2021-09-21"
+    assert first["nem_date"] == "2021-09-21T09:00:00+10:00"
+    assert first["per_kwh"] == 0.09
+    assert first["spot_per_kwh"] == 0.01
+    assert first["start_time"] == "2021-09-21T08:30:00+10:00"
+    assert first["end_time"] == "2021-09-21T09:00:00+10:00"
+    assert first["renewables"] == 50
+    assert first["spike_status"] == "none"
+    assert first["descriptor"] == "very_low"
```

Comment:
Snapshots would work beautifully here

Reply:
Using snapshots you could even parametrize the 3 tests

Reply:
Can I do this in the next PR? I've already changed more stuff than I feel comfortable with in these tests.